File: Java/Prescription.java


import java.util.Date;

public class Prescription {
    
    private Date dispenseDate = new Date();
    private int daysSupply = 30;
    
    public Prescription(Date dispenseDate, int daysSupply) {
        this.dispenseDate = dispenseDate;
        this.daysSupply = daysSupply;
    }

}

File: C++/Prescription.cpp
#include "Prescription.hpp"

#include <chrono>

using namespace std;

Prescription::Prescription(time_t dispenseDate, int daysSupply)
    : dispenseDate_(dispenseDate)
{
    expirationDate_ = chrono::system_clock::to_time_t(
        chrono::system_clock::from_time_t(dispenseDate) +
        chrono::hours(24 * daysSupply)
        );
}